crates/node/src/kitchensink.rs

```diff
@@ -0,0 +1,618 @@
+use std::{
+    io::{BufRead, Write},
+    path::PathBuf,
+    process::{Child, Command, Stdio},
+    sync::atomic::{AtomicU32, Ordering},
+    time::Duration,
+};
+
+use sc_chain_spec::{ChainSpecBuilder, ChainType};
+use serde_json::{Value, json};
+use sp_core::crypto::Ss58Codec;
+use sp_runtime::AccountId32;
+use std::fs;
+
+use alloy::{
+    hex,
+    network::EthereumWallet,
+    providers::{Provider, ProviderBuilder, ext::DebugApi},
+    rpc::types::{
+        TransactionReceipt,
+        trace::geth::{DiffMode, GethDebugTracingOptions, PreStateConfig, PreStateFrame},
+    },
+};
+
+use crate::{Node, chainspec::ChainSpec};
+use revive_dt_config::Arguments;
+use revive_dt_node_interaction::{
+    EthereumNode, trace::trace_transaction, transaction::execute_transaction,
+};
+
+static NODE_COUNT: AtomicU32 = AtomicU32::new(0);
+
+#[derive(Debug)]
+pub struct KitchensinkNode {
+    id: u32,
+    substrate_binary: PathBuf,
+    eth_proxy_binary: PathBuf,
+    rpc_url: String,
+    wallet: EthereumWallet,
+    process_substrate: Option<Child>,
+    process_proxy: Option<Child>,
+}
+
+impl KitchensinkNode {
+    const SUBSTRATE_READY_MARKER: &str = "Running JSON-RPC server";
+    const ETH_PROXY_READY_MARKER: &str = "Running JSON-RPC server";
+    const BASE_SUBSTRATE_RPC_PORT: u16 = 9944;
+    const BASE_PROXY_RPC_PORT: u16 = 8545;
+
+    fn spawn_process(&mut self, genesis: String) -> anyhow::Result<()> {
+        let substrate_rpc_port = Self::BASE_SUBSTRATE_RPC_PORT + self.id as u16;
+        let proxy_rpc_port = Self::BASE_PROXY_RPC_PORT + self.id as u16;
+
+        self.rpc_url = format!("http://127.0.0.1:{proxy_rpc_port}");
+
+        let chainspec_path = self.generate_chainspec_file(&genesis)?;
+
+        // Start Substrate node
+        let mut substrate_process = Command::new(&self.substrate_binary)
+            .arg("--chain")
```

Comment:
I think we should some more flags that `--dev` automatically would set, eg.: `--force-authoring --rpc-cors=all --alice --tmp`